Formula/pandoc.rb

```diff
@@ -1,8 +1,8 @@
 class Pandoc < Formula
   desc "Swiss-army knife of markup format conversion"
   homepage "https://pandoc.org/"
-  url "https://hackage.haskell.org/package/pandoc-2.19.2/pandoc-2.19.2.tar.gz"
-  sha256 "36e83694c36a5af35a4442c4d5abd4273289d9d309793466f59c1632e87d4245"
+  url "https://github.com/jgm/pandoc/archive/refs/tags/3.0.1.tar.gz"
+  sha256 "d8b45c3d8434c03db7cd6c571664a2a0a0ba906bfee3b8d82470103978ddc4ce"
   license "GPL-2.0-or-later"
```

Comment:
yeah, that is what I thought, but per @mjsteinbaugh's suggestion, I made the artifact location change. Happy to change to the a hackage one.

Reply:
The GitHub tarball contains all the code necessary to build both the CLI tool (pandoc-cli) and its library dependencies (pandoc, pandoc-lua-engine, pandoc-server). If you switched to the Hackage tarball for pandoc it wouldn't work, because this just has teh underlying pandoc library and not pandoc-lua-engine, pandoc-server, or pandoc-cli.  You could switch to the Hackage tarball for pandoc-cli, and build from that using `cabal v2-install -fembed_data_files`, but this would not be much different from just not downloading any tarball, and doing `cabal v2-install -fembed_data_files pandoc-cli` (which would fetch the tarball from Hackage and build).  Either way pandoc will be fetching dependent library tarballs from Hackage, so there's not much point in downloading a tarball.

Reply:
We want to be able to download and use a tarball (using `brew`, and not `cabal`) for the pandoc source for the purposes of caching and reproducibility. Is there any way we can do that? Even if we're willing to tolerate, say, `cabal` fetching tarballs for dependencies on its own?

Reply:
The `cabal.project` file has been removed, and I believe that the method of building `pandoc-cli` cabal-from a tarball should work again. If the aim is reproducibility then I'd suggest to create a [cabal freeze file](https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-freeze), which will fix all dependencies to a specific version. This is what we do for pandoc's Docker images. All packages are downloaded to the `~/.cabal/packages` folder, which might help with caching.

Reply:
What I'd recommend is that you just download the Hackage release tarball for pandoc-cli.
https://hackage.haskell.org/package/pandoc-cli-0.1/pandoc-cli-0.1.tar.gz
This is the official release tarball.

(Of course, it's kind of pointless, because you're only caching the pandoc-cli code; the huge majority of pandoc code will still be downloaded from Hackage by cabal.  But if you want to do it, it should work fine.)

You will want to use the cabal option `-f embed_data_files` when building, to ensure that the pandoc data files are baked into the binary.

Reply:
After unpacking the tarball
```
cabal install -fembed_data_files --install-method=copy --installdir=$WHERE_YOU_WANT_THE_BINARY
```